.gitignore

```diff
@@ -14,3 +14,6 @@ Cargo.lock
 
 # RPC probe results
 /rpc-probe/probe-results/
+
+# Proptest regressions dumps
+**/*.proptest-regressions
```

Comment:
Good question, I would have thought not since those are evidence of a test failing and would typically be fixed and thus obsolete before the PR gets merged. But perhaps I am understanding this wrong?

Reply:
My understanding of proptest's [failure persistence](https://altsysrq.github.io/proptest-book/proptest/failure-persistence.html) is that it re-runs any previous known failure cases before trying to generate any more novel ones, which provides additional assurance in future that the bug has been eliminated.

I haven't used proptest extensively though and may not be using it properly yet. In playing around with it today I found that it works well when you write failing tests for the first time (it persists the random seed it used to create that failure as claimed in the docs). When you then introduce a _new_ failing test, it doesn't update the regressions file for some reason.

But if you delete the regressions folder entirely and both tests fail, it saves the seeds for both failing tests :man_shrugging:

Reply:
Oh I had missed that, really neat feature actually, though it seems a bit brittle indeed in some cases.

Then it makes sense to not ignore those files and leave it it up to developers to check in these regressions in or not (one may not want to check-in a regression for a broken test).

What do you think?

Reply:
Yeah I'd say that, with property-based testing, any regressions anyone finds should be captured and committed.

Not checking it in is analogous to writing a test locally that fails on your machine (potentially showing that something's broken), but not adding it to version control, so the bug/broken code never gets found by others :grin: